tck/src/main/java/org/eclipse/microprofile/lra/tck/TckParticipantTests.java

```diff
@@ -169,7 +180,42 @@ public void testNonJaxRsCompletionStageResponseAndParticipantStatus() throws Int
 
         assertTrue("Non JAX-RS @Status method with CompletionStage<ParticipantStatus> should have been called",
             lraMetricService.getMetric(LRAMetricType.Status, lraId) >= 1);
-        
+
         lraTestService.waitForRecovery(lraId);
     }
+
+    @Test
+    public void cancelLraDuringBusinessMethod() throws InterruptedException, ExecutionException, TimeoutException {
+        LRAClientOps lraOps = lraTestService.getLRAClient();
+        URI lraId = lraOps.startLRA(null, lraClientId(), 0L, ChronoUnit.MILLIS);
+        LOGGER.info(String.format("Started LRA with URI %s", lraId));
+        // Start business method asynchronously, i.e. return immediately.
+        Future<Response> lraFuture = tckSuiteTarget.path(LongBusinessMethodParticipant.ROOT_PATH)
+                .path(LongBusinessMethodParticipant.BUSINESS_METHOD)
+                .request()
+                .header(LRA.LRA_HTTP_CONTEXT_HEADER, lraId)
+                .async()
+                .put(Entity.text(""));
+
+        // Make sure that when we cancel the LRA, the participant is waiting in the business method.
+        Response syncMethodResponse = tckSuiteTarget.path(LongBusinessMethodParticipant.ROOT_PATH)
+                      .path(LongBusinessMethodParticipant.SYNC_METHOD)
+                      .request()
+                      .put(Entity.text(""));
+        Assert.assertEquals(200, syncMethodResponse.getStatus());
+        // -1 indicates that the LRAMetricType.Compensated key is not yet present in the metrics Map.
+        // This in turn means that @Compensate could not have been called yet.
+        Assert.assertEquals(-1, lraMetricService.getMetric(
+                LRAMetricType.Compensated, lraId, LongBusinessMethodParticipant.class.getName()));
+        LOGGER.info(String.format("Cancelled LRA with URI %s", lraId));
+        lraOps.cancelLRA(lraId);
+        // waiting for the LRA to be finished
+        lraTestService.waitForRecovery(lraId);
+        // participant has to be compensated
+        Assert.assertTrue("@Compensate method should have been called at least once.",lraMetricService.getMetric(LRAMetricType.Compensated, lraId, LongBusinessMethodParticipant.class.getName()) >= 1);
```

Comment:
Should not be this formatted to two lines rather?

Reply:
Done, thanks